src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -147,6 +147,14 @@ const InfiniteScroll = ({
     }
   }, [items.length, lastPage, onMore, pendingLength, renderPageBounds, step]);
 
+  useEffect(() => {
+    if (items.length === 0 && lastPage === 0 && pendingLength !== 0) {
+      setPageHeights([]);
+      setPendingLength(0);
+      setRenderPageBounds([0, 0]);
```

Comment:
I think we should reset `renderPageBounds` consistent with how we initialize it on line 37.